lib/virtio/virtqueue.c

```diff
@@ -688,6 +796,10 @@ static int virtqueue_nused(struct virtqueue *vq)
 {
 	uint16_t used_idx, nused;
 
+#ifdef VIRTIO_D_CACHE
+	metal_cache_invalidate(VRING_USED(vq), vring_used_size(VRING_NUM(vq)));
```

Comment:
here seems that you can just invalidate vq->vq_ring.used->idx

Reply:
yeah, I'll revise everything and trying to flush / invalidate as little as possible again